### DISCUSSION TITLE
Detect passwords or empty passwords in Docker images' /etc/shadow files

### DISCUSSION THREAD

**Discussion by huornlmj** [aquasecurity/trivy#6188] [Ideas]
### Description

If a Docker image has either of the following in its /etc/shadow file then it can mean that there is either a hard coded password or an empty account password.

```
root:::0:::::
```
or

```console
root:$y$j9T$MfMQyyvhU0VftFUhpmkUS/$pPrE9qJcAx6Ac6hL5ovoMZRXoleaaex.jSZBs3iy3.1:19775:0:99999:7:::
```
Or if it is not the following:
```
root:!:::::::
```
Such an instance should be found as part of Trivy's image scanning with the secrets scanner enabled.

### Target

Container Image

### Scanner

Secret

**Comment by itaysk** [comment]
thanks for the suggestion. just to clarify, this isn't a secret (password) that was left exposed unintentionally, but rather a misconfiguration that indicates a bad practice, correct? just want to properly classify the request

**Reply by huornlmj** [reply]
It could be unintentional (accidental from the developer) or intentional (bad practice) or malicious.

**Comment by knqyf263** [comment]
Same request?
https://github.com/aquasecurity/trivy/pull/3798

**Reply by huornlmj** [reply]
No I don't think this is the same as a weak password specifically for SSH.  It's more broad than that.

**Reply by knqyf263** [reply]
It's not limited to SSH. The PR tries to scan `/etc/shadow,` `/etc/disc.conf`, etc., for weak passwords.

**Reply by huornlmj** [reply]
But will it detect `root:::0:::::` as weak?

**Reply by knqyf263** [reply]
I'm not sure if weak passwords include an empty one or not.
